csrc/class_featureset.cpp

```diff
@@ -57,10 +57,12 @@ JNIEXPORT jobjectArray JNICALL Java_mapnik_FeatureSet__1loadGeometries
 		return 0;
 	}
 
-	unsigned count=(*fp)->num_geometries();
+	// Mapnik 3.x only holds one geometry<> per feature_impl.
+	// (In Mapnik 2.x this used to be a variable-length vector of geometries.)
+	unsigned count=1;
```

Comment:
If I didn't misunderstand things here, the JNI interface should be refactored to match the new C++ interface - but I didn't want to expand the scope of this PR for something that I don't really understand (I only want to render maps :)).